virttest/qemu_vm.py

```diff
@@ -2560,6 +2577,20 @@ def sort_key(dev):
             add_balloon(devices, devid=balloon_devid, bus=balloon_bus,
                         use_old_format=use_ofmt, options=options)
 
+        for virtio_mem_device in params.objects("virtio_mem"):
+            virtio_mem_params = params.object_params(virtio_mem_device)
+            options = {}
+            options["id"] = virtio_mem_device
+            requested_size = virtio_mem_params.get("virtio_mem_requested_size")
+            options["requested-size"] = requested_size
+            memdev = virtio_mem_params.get("virtio_mem_memdev")
+            options["memdev"] = memdev
+            node = virtio_mem_params.get("virtio_mem_node")
+            options["node"] = node
+            bus = virtio_mem_params.get("virtio_mem_bus")
+            options["bus"] = bus
```

Comment:
An important thing is: `bus`  should be set with `parent_bus=bus` in `QDevice` and it should not be defined by users, otherwise `QPCI(E)Bus` could not generate a correct address.

Reply:
@PaulYuuu so how can I set to the device which bus will I use?

Reply:
https://github.com/avocado-framework/avocado-vt/blob/6e92a55fd4ce018df49d5bbf77bf994377ada0fd/virttest/qemu_vm.py?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L1468

https://github.com/avocado-framework/avocado-vt/blob/6e92a55fd4ce018df49d5bbf77bf994377ada0fd/virttest/qemu_vm.py?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L3758-L3787